cli/cmd/upgrade_test.go

```diff
@@ -202,6 +204,13 @@ func TestUpgradeOverwriteIssuer(t *testing.T) {
 				// Trust root has changed.
 				continue
 			}
+
+			if id == "Deployment/linkerd-identity" || id == "Deployment/linkerd-proxy-injector" {
+				if !pathMatch(diff.path, []string{"spec", "template", "spec", "containers", "*", "args", "*"}) || strings.TrimPrefix(diff.b.(string), "-identity-trust-anchors-pem=") != issuerCerts.ca {
```

Comment:
- If there's a diff outside the container args then fail
- If there's a diff in some of the containers arg, check if that arg's value differs from `issuerCerts.ca`. What if the diff is not for that particular arg?